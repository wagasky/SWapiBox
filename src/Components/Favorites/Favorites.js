import React from 'react';
import Card from '../Card/Card'

const Favorites = ({ data, handleFavorite }) => {

  const renderedCards = data.map((favorite, index) => {
    return (
      <Card 
        key={index}
        data={favorite}
        handleFavorite={handleFavorite}
      />
    )
  })

  return (
    <div className="favoriteCards">
      { renderedCards }
    </div>
  )
}

export default Favorites;