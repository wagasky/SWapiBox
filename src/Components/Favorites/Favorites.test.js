/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import Favorites from './Favorites';
import { shallow } from 'enzyme';

it.skip('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Favorites />, div);
  ReactDOM.unmountComponentAtNode(div);
});