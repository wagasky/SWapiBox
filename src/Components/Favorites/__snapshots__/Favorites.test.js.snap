// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot with 1 card 1`] = `
<div
  className="favorite-cards"
>
  <Card
    data={
      Object {
        "homeworld": "Tatooine",
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
    key="0"
  />
</div>
`;

exports[`Favorites should match snapshot with 3 cards 1`] = `
<div
  className="favorite-cards"
>
  <Card
    data={
      Object {
        "homeworld": "Tatooine",
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
    key="0"
  />
  <Card
    data={
      Object {
        "climate": "temperate",
        "name": "Alderaan",
        "population": "2000000000",
        "residents": "Leia Organa
      Bail Prestor Organa
      Raymus Antilles",
        "terrain": "grasslands, mountains",
      }
    }
    key="1"
  />
  <Card
    data={
      Object {
        "model": "t-47 airspeeder",
        "name": "Snowspeeder",
        "passengers": "0",
        "vehicleClass": "airspeeder",
      }
    }
    key="2"
  />
</div>
`;
